src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -490,10 +510,12 @@ public void Dispose()
             _certContext = null;
             if (certContext != null && !certContext.IsInvalid)
             {
-                if (!_certContextCloned)
+                if (_certContextCloned)
                 {
-                    certContext.Dispose();
+                    certContext.DangerousRelease();
```

Comment:
This will only get called if this object gets Disposed.  The DangerousRelease needs to come from something with a finalizer.

The way this is floating in my head is

```C#
internal SafeCertContextHandle(SafeCertContextHandle parent)
{
    bool ignored = false;
    parent.DangerousAddRef(ref ignored);
    _parent = parent;
    SetHandle(_parent.handle);
}

protected override bool ReleaseHandle()
{
    if (_parentHandle != null)
    {
        _parentHandle.DangerousRelease();
        _parentHandle = null;
    }
    else
    {
        Interop.crypt32.CertFreeCertificateContext(handle);
    }

    SetHandle(IntPtr.Zero);
    return true;
}
```

Then you'd just do `_certContext = new SafeCertContextHandle(copyFrom._certContext);` in the ctor.